### ISSUE TITLE
Artist page loading

### ISSUE THREAD

**Issue by azukaar** [navidrome/navidrome#2222]
### Description

I was wondering if the way images for artists changed recently because the artist pages are now broken through the SoniXD player.

![image](https://user-images.githubusercontent.com/7872597/222748742-00da7a6f-445a-4478-b19c-ee55dda09fe3.png)

I dont really mind the artist list not being pretty, but what is a problem, is that the artist pages once you click on the artist with the broken image logo, they don't load at all (infinite loading).

On the navidrome web page, when I try to access an artist with the default "star shaped" image, it's fine, but when I try to access some other with the broken icon image, on the Navidrome web client htey have an empty image 

![image](https://user-images.githubusercontent.com/7872597/222750017-5d9ba24f-efd3-4c94-98c3-3d7c400efefd.png)

And if I look at the link I have 

```
/rest/getCoverArt?u=yann&t=b08d65bed3969c7be10f47d8e06a410f&s=614d01&f=json&v=1.8.0&c=NavidromeUI&id=ar-addeb336f8da2ba4f13fedabed3cf917&size=300
```

Which returns an xml instead of an image

```
<artist>
  <name>DJ Contacreast</name>
  <musicbrainzartistid>b23b473a-2107-4d34-b163-cf0ac475cd93</musicbrainzartistid>
  <biography></biography>
  <outline></outline>
</artist>
```

Which obviously shouldn\'t be the output of the /rest/getCoverArt route

 - Navidrome version: 0.49.3
 - SoniXD: 0.15.4

**Comment by deluan** [comment]
Thanks for reporting. I've seen this issue happening once or twice, it takes a long time to resolve images (> 2 minutes) eventually timing out maybe even polluting the cache. I don't know exactly why this happens, but the xml you see as a response to that endpoint does not make sense at all. I'll try to investigate this a bit more.

If anyone has any ideas, please let me know.

**Comment by stasadance** [comment]
> Thanks for reporting. I've seen this issue happening once or twice, it takes a long time to resolve images (> 2 minutes) eventually timing out maybe even polluting the cache. I don't know exactly why this happens, but the xml you see as a response to that endpoint does not make sense at all. I'll try to investigate this a bit more.
> 
> If anyone has any ideas, please let me know.

After some experimenting I found out that this issue is caused by Lidarr's method of storing metadata in the artist folder.

The reason for the XML output is the only file that contains "artist" in the folder is usually "artist.nfo"

The artist image is usually in a file called "folder.jpg"

If you rename this file to "artist.jpg" it will load in Navidrome correctly.

**Comment by azukaar** [comment]
But this has to be something that changed recently because I have been using this server for a year

**Comment by deluan** [comment]
> But this has to be something that changed recently because I have been using this server for a year

This changed in 0.49.0, with the new [Artwork location resolution](https://www.navidrome.org/docs/usage/artwork/#artists).

The next version will allow you to customize the artist images filenames: https://github.com/navidrome/navidrome/issues/2151

@azukaar, can you confirm this is the issue you were facing, so we can close this?

**Comment by azukaar** [comment]
I renamed, and quick-scanned one, but no avail.. I'm not sure if I did it wrong? URL still returns an xml

![image](https://user-images.githubusercontent.com/7872597/229364317-82986a3a-fcd0-4b95-b2c2-2c9da4966c51.png)

Also what was doing before 0.49.0 that was working? Was it just displaying any images?

**Comment by CappielloAntonio** [comment]
Same situation, same problem. The problem is solved by updating (temporarily, until the fix is reported in the stable) to the latest developer version, and setting the ArtistArtPriority parameter in my environment variables with the following order 'folder.*, artist.*, album /artist.*, external'. No rescan required, the effects are immediate.

![Screenshot_20230402-180814](https://user-images.githubusercontent.com/14802539/229365063-bd3bc371-2013-46d1-88c3-6eb0f38525c4.png)

**Comment by deluan** [comment]
> Also what was doing before 0.49.0 that was working? Was it just displaying any images?

Before 0.49.0, Navidrome always tried to get images from Spotify if configured. If not, it would only display the "gray star".

The only current workarounds before the next release are:
1. Remove the `artist.xml` files from your library
2. As suggested by @CappielloAntonio, you can use the latest development build, that includes the new `ArtistArtPriority` config option.

**Comment by azukaar** [comment]
Thanks, I definitly get the part where I have 2 options to display the images.
But I think this ticket still stands on its own as a valid issue, as the new image retrieval system should filter extensions that ARE images (png, jpg, etc...) and not serves random xml files as images, no? (And then show the gray star if none is found)
Because pre-0.49 I didn't have artist images but it was fine, because at least the artist page were working, but serving xml as images is breaking clients such as Sonixd, and the artist pages are not loading anymore. That's a regression no?

**Comment by CappielloAntonio** [comment]
At the [getCoverArt](http://www.subsonic.org/pages/api.jsp#getCoverArt) API call right now Navidrome returning the artist.* file, whatever its extension (in your case, due to Lidarr, xml).

The next version of Navidrome will allow us to customize the filename that Navidrome will prioritize for the artist cover *file* (not necessarily an image, but a file).

To better understand Navidrome's current behavior, here is an extract from the [official documentation](https://www.navidrome.org/docs/usage/artwork/#artists):

- Try to read an artist.* image file from the artist folder.

- If not found, try to read an artist.* image from any album for that artist. If more than one is found, the selected one is unpredictable.

- If not found, it will try to get it from an external source. Currently, this can only be Spotify, if configured.

- If not found, use the artist image placeholder (grey star image).

Coming back to your specific case, the next version of Navidrome won't change the way we respond to the standard Subsonic API, as it will give us more flexibility in choosing the source file to use.

**Comment by azukaar** [comment]
Yes I definitely understood this, but as per my comment, and I don't want to seem insisting or anything, I still think that returning artist.* instead of artist.[png/jpg/..] is a regression since it breaks out-of-box compatibility with very common tools and returning an XML to a /getImage route is definitely an unexpected behavior.
Changing the default from artist.* to artist.[png/jpg/..] and keeping everything else the same (with your new customizable stuff) would fix the regression and keep the new customizable behaviour
But up to you, feel free to close the ticket if you disagree

**Comment by deluan** [comment]
@azukaar I kept `artist.*` as default, but only now it consider files that are actually images (by extension). Fixed in 6a8d2dc8.

I'll close this as the issue is resolved, but feel free to chime in if you have any other concerns.

**Comment by azukaar** [comment]
Fantastic, BTW I know it's not the place, but I have been using Navidrome for probably over a year now, and I just want to add an appreciation comment to this thread, so... Honestly, good job! And thanks
